packages/hdwallet-keepkey/src/binance.ts

```diff
@@ -45,16 +45,22 @@ export async function binanceSignTx(
     );
     if (resp.message_type === Core.Events.FAILURE) throw resp;
 
+    const intRegex = /^(0|([1-9][0-9]*))$/;
```

Comment:
If we're importing `bignumber.js` we might as well use their already tested function to determine in the value is a whole number

https://mikemcl.github.io/bignumber.js/#isInt

Reply:
if we make it available in hdwallet-keepkey i agree. was trying to optimize to import in the smallest scope possible; didn't want to import a whole lib just to get that function. OTOH we could for example `bignumber.js` in the workspace deps, and use it here for no additional cost. Thoughts?

Reply:
`bignumber.js` is a very small package but I understand the desire to not include another dependency. In our case, since we already use `bignumber.js` yarn/npm will hoist it resulting in no need for an additional copy of the dependency to be installed.

In this case I'm leaning more toward using the package as it has been well tested/hardened and less likely to have a gotcha.

However, if you create some tests for this code that covers all kinds of invalid inputs and you can validate that it works as expected, then I would be okay with that (though I would recommend refactoring out into a shared helper function).

Reply:
i'd prefer the workspace-root approach myself, just didn't have enough background knowledge to feel comfortable doing it that way. will refactor and rebase.

Reply:
sigh -- we have a lot of dev deps at the workspace level, but no non-dev ones. i'm guessing that's for a good reason, b/c i don't think the eventually-published npm packages will include workspace-level deps. I'll just add `bignumber.js` to both packages instead.